CONTRIBUTING.md

```diff
@@ -42,11 +64,63 @@ uv sync --extra text --extra video
 uv sync --extra all
 ```
 
-### Dev Pattern
-
-- Sign and signoff commits with `git commit -sS`. (May be relaxed in the future)
 - If project dependencies are updated a new uv lock file needs to be generated. Run `uv lock` and add the changes of the new uv.lock file.
 
-### Testing
+## Unit tests
+Unit tests should be simple and fast.
+Developers should be able to run them frequently while developing without any slowdown.
+```
+pytest
+# If you don't have NVIDIA GPU do:
+# pytest -m 'not gpu'
+```
+
+## Coverage
+Pull requests should cover at least 80% of its changes with tests. CI will reject PRs that do not fulfill this requirement. Please refer to the [Unit tests](#unit-tests) section for more about writing unit tests.
+
+## Pull Requests (PR) Guidelines
+
+**Send your PRs to the `main` branch**
+
+1) Make sure your PR does one thing. Have a clear answer to "What does this PR do?".
+2) Read General Principles and style guide above
+3) Make sure you sign your commits. E.g. use ``git commit -sS`` when committing.
+    1) If you forget to do this, please follow the steps below to undo the commits and reapply the changes under a new (signed and signed-off) commit. Note: This will preserve your changes, but delete the git history of commits.
+    ```bash
+    git reset --soft HEAD~N
+    git add <insert all files you want to include>
+    git commit -sS -m "My commit message"
+    git push --force
+    ```
+    Replace `N` in the first line with the number of commits you want to undo. To undo the latest commit, do `git reset --soft HEAD~1`.
```

Comment:
Can we link to some page for how to set up signature verification? I think folks often struggle with that step too.